### ISSUE TITLE
Limited width display in Python Interactive window despite settings

### ISSUE THREAD

**Issue by Tirbo06** [microsoft/vscode-jupyter#8053]
## Environment data

-   VS Code version: 1.61.2
-   Extension version (available under the Extensions sidebar):  **Python v2021.10.1365161279**
-   OS and version: **Windows 10 21H1**
-   Python version (& distribution if applicable, e.g. Anaconda): **3.8.1 / 3.9**
-   Relevant/affected Python packages and their versions: **Interactive Windows (python)**
-   Value of the `python.languageServer` setting: **PyLance**

## Expected behaviour

Display large width results (e.g text) in the interactive window python,  

## Actual behaviour

Split at the half of the width of the interactive window

## Steps to reproduce:

Display large text in the interactive python window

## What I have tried
- find informations 
- set a big value for pandas.options.display.max_columns
- set a big value for notebook.output.textLineLimit
- seen issue https://github.com/microsoft/vscode-python/pull/6518

Here is a picture with the bad output width and the modified setting
![display_width_output](https://user-images.githubusercontent.com/59821431/138870347-8952dcef-b766-46e1-8d86-ace107f881f4.jpg)

</p>
</details>

**Comment by Tirbo06** [comment]
no pb 

```
import os
import numpy as np 
import pandas as pd 

data = pd.read_json('../data/frames.json')


turn = data.turns[0]
print('Number of items in 1 turn example: ', len(turn), '\n-----------------------------------------------------')
print(data['labels'])

data
```


Here is the start of the file *frames.json*, you can DL here : 
https://www.microsoft.com/en-us/research/project/frames-dataset/#download


```
[{"user_id": "U22HTHYNP", "turns": [{"text": "I'd like to book a trip to Atlantis from Caprica on Saturday, August 13, 2016 for 8 adults. I have a tight budget of 1700.", "labels": {"acts": [{"args": [{"val": "book", "key": "intent"}], "name": "inform"}, {"args": [{"val": "Atlantis", "key": "dst_city"}, {"val": "Caprica", "key": "or_city"}, {"val": "Saturday, August 13, 2016", "key": "str_date"}, {"val": "8", "key": "n_adults"}, {"val": "1700", "key": "budget"}], "name": "inform"}], "acts_without_refs": [{"args": [{"val": "book", "key": "intent"}], "name": "inform"}, {"args": [{"val": "Atlantis", "key": "dst_city"}, {"val": "Caprica", "key": "or_city"}, {"val": "Saturday, August 13, 2016", "key": "str_date"}, {"val": "8", "key": "n_adults"}, {"val": "1700", "key": "budget"}], "name": "inform"}], "active_frame": 1, "frames": [{"info": {"intent": [{"val": "book", "negated": false}], "budget": [{"val": "1700.0", "negated": false}], "dst_city": [{"val": "Atlantis", "negated": false}], "or_city": [{"val": "Caprica", "negated": false}], "str_date": [{"val": "august 13", "negated": false}], "n_adults": [{"val": "8", "negated": false}]}, "frame_id": 1, "requests": [], "frame_parent_id": null, "binary_questions": [], "compare_requests": []}]}, "author": "user", "timestamp": 1471272019730.0}, {"db": {"result": [[{"trip": {"returning": {"duration": {"hours": 0, "min": 51}, "arrival": {"hour": 10, "year": 2016, "day": 24, "min": 51, "month": 8}, "departure": {"hour": 10, "year": 2016, "day": 24, "min": 0, "month": 8}}, "seat": "ECONOMY", "leaving": {"duration": {"hours": 0, "min": 51}, "arrival": {"hour": 0, "year": 2016, "day": 16, "min": 51, "month": 8}, "departure": {"hour": 0, "year": 2016, "day": 16, "min": 0, "month": 8}}, "or_city": "Porto Alegre", "duration_days": 9}, "price": 2118.81, "hotel": {"gst_rating": 7.15, "vicinity": [], "name": "Scarlet Palms Resort", "country": "Brazil", "amenities": ["FREE_BREAKFAST", "FREE_PARKING", "FREE_WIFI"], "dst_city": "Goiania", "category": "3.5 star hotel"}}, {"trip": {"returning": {"duration": {"hours": 2, "min": 37}, "arrival": {"hour": 12, "year": 2016, "day": 10, "min": 37, "month": 8}, "departure": {"hour": 10, "year": 2016, "day": 10, "min": 0, "month": 8}}, "seat": "ECONOMY", "leaving": {"duration": {"hours": 2, "min": 37}, "arrival": {"hour": 0, "year": 2016, "day": 4, "min": 37, "month": 8}, "departure": {"hour": 22, "year": 2016, "day": 3, "min": 0, "month": 8}}, "or_city": "Porto Alegre", "duration_days": 7}, "price": 2369.83, "hotel": {"gst_rating": 0, "vicinity": [], "name": "Sunway Hostel", "country": "Argentina", "amenities": ["FREE_BREAKFAST", "FREE_WIFI"], "dst_city": "Rosario", "category": "2.0 star hotel"}}, {"trip": {"returning": {"duration": {"hours": 0, "min": 51}, "arrival": {"hour": 10, "year": 2016, "day": 24, "min": 51, "month": 8}, "departure": {"hour": 10, "year": 2016, "day": 24, "min": 0, "month": 8}}, "seat": "BUSINESS", "leaving": {"duration": {"hours": 0, "min": 51}, "arrival": {"hour": 0, "year": 2016, "day": 16, "min": 51, "month": 8}, "departure": {"hour": 0, "year": 2016, "day": 16, "min": 0, "month": 8}}, "or_city": "Porto Alegre", "duration_days": 9}, "price": 2375.72, "hotel": {"gst_rating": 7.15, "vicinity": [], "name": "Scarlet Palms Resort", "country": "Brazil", "amenities": ["FREE_BREAKFAST", "FREE_PARKING", "FREE_WIFI"], "dst_city": "Goiania", "category": "3.5 star hotel"}}, {"trip": {"returning": {"duration": {"hours": 1, "min": 30}, "arrival": {"hour": 11, "year": 2016, "day": 1, "min": 30, "month": 9}, "departure": {"hour": 10, "year": 2016, "day": 1, "min": 0, "month": 9}}, "seat": "BUSINESS", "leaving": {"duration": {"hours": 1, "min": 30}, "arrival": {"hour": 18, "year": 2016, "day": 19, "min": 30, "month": 8}, "departure": {"hour": 17, "year": 2016, "day": 19, "min": 0, "month": 8}}, "or_city": "Porto Alegre", "duration_days": 13}, "price": 2492.95, "hotel": {"gst_rating": 0, "vicinity": [], "name": "Hotel Mundo", "country": "Brazil", "amenities": ["FREE_BREAKFAST", "FREE_WIFI", "FREE_PARKING"], "dst_city": "Manaus", "category": "2.5 star hotel"}}, {"trip": {"returning": {"duration": {"hours": 0, "min": 51}, "arrival": {"hour": 10, "year": 2016, "day": 31, "min": 51, "month": 8}, "departure": {"hour": 10, "year": 2016, "day": 31, "min": 0, "month": 8}}, "seat": "ECONOMY", "leaving": {"duration": {"hours": 0, "min": 51}, "arrival": {"hour": 19, "year": 2016, "day": 27, "min": 51, "month": 8}, "departure": {"hour": 19, "year": 2016, "day": 27, "min": 0, "month": 8}}, "or_city": "Porto Alegre", "duration_days": 4}, "price": 2538.0, "hotel": {"gst_rating": 8.22, "vicinity": [], "name": "The Glee", "country": "Brazil", "amenities": ["FREE_BREAKFAST", "FREE_WIFI"], "dst_city": "Recife", "category": "4.0 star hotel"}}], [], [], [], [], [], []], "search": [{"ORIGIN_CITY": "Porto Alegre", "PRICE_MIN": "2000", "NUM_ADULTS": "2", "timestamp": 1471271949.995, "PRICE_MAX": "3000", "ARE_DATES_FLEXIBLE": "true", "NUM_CHILDREN": "5", "START_TIME": "1470110400000", "MAX_DURATION": 2592000000.0, "DESTINATION_CITY": "Brazil", "RESULT_LIMIT": "10", "END_TIME": "1472616000000"}, {"ORIGIN_CITY": "Atlantis", "NUM_ADULTS": "8", "RESULT_LIMIT": "10", "timestamp": 1471272148.124, "PRICE_MAX": "1700", "NUM_CHILDREN": "", "ARE_DATES_FLEXIBLE": "true", "START_TIME": "NaN", "END_TIME": "NaN"}, {"ORIGIN_CITY": "Caprica", "PRICE_MAX": "1700", "NUM_ADULTS": "8", "RESULT_LIMIT": "10", "timestamp": 1471272189.07, "DESTINATION_CITY": "Atlantis", "NUM_CHILDREN": "", "ARE_DATES_FLEXIBLE": "true", "START_TIME": "1470715200000", "END_TIME": "1472011200000"}, {"ORIGIN_CITY": "Caprica", "PRICE_MAX": "1700", "NUM_ADULTS": "8", "RESULT_LIMIT": "10", "timestamp": 1471272205.436, "DESTINATION_CITY": "Atlantis", "NUM_CHILDREN": "", "ARE_DATES_FLEXIBLE": "true", "START_TIME": "1470715200000", "END_TIME": "1472011200000"}, {"ORIGIN_CITY": "Caprica", "PRICE_MIN": "1700", "NUM_ADULTS": "8", "RESULT_LIMIT": "10", "timestamp": 1471272278.72, "DESTINATION_CITY": "Atlantis", "NUM_CHILDREN": "", "ARE_DATES_FLEXIBLE": "true", "START_TIME": "1470715200000", "END_TIME": "1472011200000"}, {"ORIGIN_CITY": "Caprica", "PRICE_MIN": "1700", "NUM_ADULTS": "8", "RESULT_LIMIT": "10", "timestamp": 1471272454.542, "DESTINATION_CITY": "Atlantis", "NUM_CHILDREN": "", "ARE_DATES_FLEXIBLE": "true", "START_TIME": "1471060800000", "END_TIME": "1472011200000"}, {"ORIGIN_CITY": "Caprica", "PRICE_MIN": "1700", "NUM_ADULTS": "8", "RESULT_LIMIT": "10", "timestamp": 1471272466.008, "DESTINATION_CITY": "Atlantis", "NUM_CHILDREN": "", "ARE_DATES_FLEXIBLE": "true", "START_TIME": "1471060800000", "END_TIME": "1472011200000"}]}, "text": "Hi...I checked a few options for you, and unfortunately, we do not currently have any trips that meet this criteria.  Would you like to book an alternate travel option?", "labels": {"acts": [{"args": [{"val": [{"annotations": [], "frame": 1, "fromrange": false}], "key": "ref"}], "name": "no_result"}, {"args": [{"key": "dst_city"}], "name": "suggest"}, {"args": [], "name": "sorry"}], "active_frame": 1, "frames": [{"info": {"intent": [{"val": "book", "negated": false}], "budget": [{"val": "1700.0", "negated": false}], "dst_city": [{"val": "Atlantis", "negated": false}], "or_city": [{"val": "Caprica", "negated": false}], "str_date": [{"val": "august 13", "negated": false}], "n_adults": [{"val": "8", "negated": false}], "NO_RESULT": [{"val": true, "negated": false}]}, "frame_id": 1, "requests": [], "frame_parent_id": null, "binary_questions": [], "compare_requests": []}]}, "author": "wizard", "timestamp": 1471272381399.0}, {"text": "Yes, how about going to Neverland from Caprica on August 13, 2016 for 5 adults. For this trip, my budget would be 1900.", "labels": {"acts": [{"args": [{"val": "Neverland", "key": "dst_city"}, {"val": "5", "key": "n_adults"}, {"val": "1900", "key": "budget"}, {"val": [{"annotations": [{"val": "Caprica", "key": "or_city"}, {"val": "August 13, 2016", "key": "str_date"}], "frame": 1, "fromrange": false}], "key": "ref"}], "name": "inform"}], "acts_without_refs": [{"args": [{"val": "Neverland", "key": "dst_city"}, {"val": "5", "key": "n_adults"}, {"val": "1900", "key": "budget"}, {"val": "Caprica", "key": "or_city"}, {"val": "August 13, 2016", "key": "str_date"}], "name": "inform"}], "active_frame": 2, "frames": [{"info": {"intent": [{"val": "book", "negated": false}], "budget": [{"val": "1700.0", "negated": false}], "dst_city": [{"val": "Atlantis", "negated": false}], "or_city": [{"val": "Caprica", "negated": false}], "str_date": [{"val": "august 13", "negated": false}], "n_adults": [{"val": "8", "negated": false}], "NO_RESULT": [{"val": true, "negated": false}]}, "frame_id": 1, "requests": [], "frame_parent_id": null, "binary_questions": [], "compare_requests": []}, {"info": {"intent": [{"val": "book", "negated": false}], "budget": [{"val": "1900.0", "negated": false}], "dst_city": [{"val": "Neverland", "negated": false}], "or_city": [{"val": "Caprica", "negated": false}], "str_date": [{"val": "august 13", "negated": false}], "n_adults": [{"val": "5", "negated": false}]}, "frame_id": 2, "requests": [], "frame_parent_id": 1, "binary_questions": [], "compare_requests": []}]}, "author": "user", "timestamp": 1471272508882.0}, {"db": {"result": [[], [], [], [], [], []], "search": [{"ORIGIN_CITY": "Caprica", "PRICE_MIN": "1700", "NUM_ADULTS": "8", "RESULT_LIMIT": "10", "timestamp": 1471272454.542, "DESTINATION_CITY": "Atlantis", "NUM_CHILDREN": "", "ARE_DATES_FLEXIBLE": "true", "START_TIME": "1471060800000", "END_TIME": "1472011200000"}, {"ORIGIN_CITY": "Caprica", "PRICE_MIN": "1700", "NUM_ADULTS": "8", "RESULT_LIMIT": "10", "timestamp": 1471272466.008, "DESTINATION_CITY": "Atlantis", "NUM_CHILDREN": "", "ARE_DATES_FLEXIBLE":  ....
```

**Comment by rchiodo** [comment]
Yes this is a limitation of pandas. It's returning that text (the same thing repros in a jupyter notebook).

![image](https://user-images.githubusercontent.com/19672699/139723256-04371771-6242-4aad-969e-0b11ab38b2d8.png)


However our dataframe viewer should let you see these objects. It isn't:

![image](https://user-images.githubusercontent.com/19672699/139723357-fdfe7a80-d76b-4c50-87ed-a48669963165.png)

That's a bug that we should fix. This issue actually https://github.com/microsoft/vscode-jupyter/issues/7915. 

So theoretically when we fix #7915, you'll be able to see the labels column in your data.

**Comment by Tirbo06** [comment]
Ok thank you !
